src/containers/Widgets/Widgets.js

```diff
@@ -28,7 +28,10 @@ export default class Widgets extends Component {
   static reduxAsyncConnect(params, store) {
     const {dispatch, getState} = store;
     if (!isLoaded(getState())) {
-      return dispatch(loadWidgets());
+      const promise = dispatch(loadWidgets());
+      if (__SERVER__) {
```

Comment:
It seems for me, that logic of fetchDataDeferred was not really obvious... 
Of course, I can add new static method , analogue of fetchDataDeferred, but i'd like to find more elegant solution...